src/libs/web3-data-provider/Web3Provider.tsx

```diff
@@ -113,7 +113,7 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
       // close will remove wallet from DOM if provided by wallet
       await connector.close();
     }
-
+    setWalletType(undefined);
```

Comment:
@0x4Graham what is this for?

Reply:
@grothem suggested we do this, so that on disconnect we clear the wallet type so that when they reconnected we get the new wallet type